src/cli.rs

```diff
@@ -450,18 +505,25 @@ impl Opts {
     async fn update(&self, opts: &UpdateOpts) -> Result<()> {
         let mut pins = self.read_pins()?;
 
+        let strategy = match (opts.partial, opts.full) {
+            (false, false) => UpdateStrategy::Normal,
+            (false, true) => UpdateStrategy::Full,
+            (true, false) => UpdateStrategy::HashesOnly,
+            (true, true) => panic!("partial and full are mutually exclusive"),
```

Comment:
Panic isn't really a great way of handling this. Can we properly log this with an exit code of 1?

Reply:
The user can't specify both; the CLI arument parse will take care of this: "error: The argument '--partial' cannot be used with '--full'". On the other hand, programmer mistakes should fail loudly.

Reply:
Moved to #28